source/server/admin/stats_handler.cc

```diff
@@ -187,8 +198,21 @@ void StatsHandler::statsAsText(const std::map<std::string, uint64_t>& all_stats,
   std::map<std::string, std::string> all_histograms;
   for (const Stats::ParentHistogramSharedPtr& histogram : histograms) {
     if (shouldShowMetric(*histogram, used_only, regex)) {
-      auto insert = all_histograms.emplace(histogram->name(), histogram->quantileSummary());
-      ASSERT(insert.second); // No duplicates expected.
+      bool success = false;
+      // Display bucket data if histogram_buckets query parameter is used, otherwise output contains
+      // quantile summary data.
+      if (!histogram_buckets_value.has_value()) {
+        success = all_histograms.emplace(histogram->name(), histogram->quantileSummary()).second;
+      } else if (histogram_buckets_value.value() == "cumulative") {
+        success = all_histograms.emplace(histogram->name(), histogram->bucketSummary()).second;
+      } else {
+        // "disjoint" should be only possible value at this point. Other values should already have
+        // caused a bad request response.
+        ASSERT(histogram_buckets_value.value() == "disjoint");
```

Comment:
Can you make an enum for the possible values, use a switch-statement rather than if/else-if block, and drop the assert?

You can put the error message at the place where you parse the query-param into the enum.